lib/layout.js

```diff
@@ -615,6 +618,34 @@ export default class PodiumLayout {
                 res.podiumSend = (data, ...args) =>
                     res.send(this.render(incoming, data, ...args));
 
+                // we add a new podiumStream method that sets up streaming and returns the stream
+                // object for the developer to work with
+                res.podiumStream = (...args) => {
+                    const responseStream = new ResponseStream();
+                    // pipe the readable response stream into the express writeable res stream
+                    // to set up streaming
+                    responseStream.pipe(res);
+
+                    // call our document template, injecting a token we can use to split the template
+                    // into header and footer
+                    const splitToken = `<!--split-token-->`;
```

Comment:
I'm having a tough time visualizing how this split would affect loading. Is this only here to handle the `server-simple` case of shipping off `<head>` and `<body>` in two chunks?

Reply:
We split the template on the position that would normally have the body injected by the developers so that we are left with 2 pieces that we are able to deliver. The first bit goes immediately, then we allow the user to stream whatever they want, and then we deliver the second bit at the end. In practice this looks like...

We deliver this right away..

```js
<html>
<head>
...
</head>
<body>
```

Then the developer does there thing and streams the content they want in the body... podlets etc

```js
<main>
... podlets etc...
</main>
```

Then we deliver the second split part of the template which is the document end...

```js
... some scripts etc...
</body>
</html>
```

Reply:
The nice thing about using the split technique is that all existing templates should just work as they are so its nice and backwards compatible

Reply:
I guess my main concern about shipping a "finished" `<head>` is that podlets' early hints might give us styles and scripts that should be up there. I suppose that's a design constraint with streaming you have to deal with as an app developer, that your stuff can come in at any time?

Reply:
we wait for all assets before shipping the head by using incoming.hints.on('complete')